### PR TITLE
[JENKINS-39570] Do not display maven modules in pipeline listing

### PR DESCRIPTION
**Pull request by imeredith**
# Description

Removes maven modules from pipeline listing.

Tests done in backend not with ATH.

See [JENKINS-39570](https://issues.jenkins-ci.org/browse/JENKINS-39570).

@michaelneale can you please take a look and make sure its what you expect to happen?

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
code looks good - will have to try it out later, nice one.

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java [hunk 1]
```diff
@@ -280,6 +280,12 @@ public Job getJob(){
 
         @Override
         public BluePipeline getPipeline(Item item, Reachable parent) {
+            // Maven modules are not pipelines.
+            // TODO: Handle Maven modules better?
+            if (item != null && item.getClass().getName().equals("hudson.maven.MavenModule")) {
+                return null;
```
Comment (imeredith) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/AbstractPipelineImpl.java]: No, they are AbstractProjects.

**Comment by michaelneale** [comment]
@imeredith sorry I can't work out how to do a maven build - have installed plugin etc as per ticket, but it won't work (not related to blue ocean) so I am not sure if this works as expected.

**Comment by michaelneale** [comment]
ok managed to mash the keyboard and make it work. 
go for it :bee:

**Pull request merged**